### ISSUE TITLE
AutoComplete: panel height should fit items when using virtual scrolling

### ISSUE THREAD

**Issue by inad9300** [primefaces/primereact#7237]
### Describe the bug

Enabling virtual scrolling in `<AutoComplete />` has the following undesired side effect:
![image](https://github.com/user-attachments/assets/0729c912-d72f-4d14-8ba1-2d407ae73bf7)

Compare with the behavior without virtual scrolling:
![image](https://github.com/user-attachments/assets/2847079e-7ca1-4b0e-a257-29916b1be200)

### Reproducer

https://primereact.org/autocomplete/#virtualscroll

### System Information

```Shell
System:
    OS: Linux 6.8 Ubuntu 22.04.5 LTS 22.04.5 LTS (Jammy Jellyfish)
    CPU: (8) x64 11th Gen Intel(R) Core(TM) i7-1165G7 @ 2.80GHz
    Memory: 2.24 GB / 15.40 GB
    Container: Yes
    Shell: 5.1.16 - /bin/bash
  Binaries:
    Node: 20.11.1 - ~/.nvm/versions/node/v20.11.1/bin/node
    npm: 10.2.4 - ~/.nvm/versions/node/v20.11.1/bin/npm
  Browsers:
    Chrome: 129.0.6668.58
  npmPackages:
    primereact: 10.6.6 => 10.8.3 
    react: 18.2.0 => 18.2.0
```

**Comment by blan19** [comment]
AutoComplete 컴포넌트의 Virtual Scroll은 내부적으로 VirtualScroller 컴포넌트를 사용하고 있습니다.

```typescript
if (props.virtualScrollerOptions) {
  const items = props.suggestions ? (props.optionGroupLabel ? flattenGroupedItems(props?.suggestions) : props.suggestions) : null;

  const virtualScrollerProps = {
    ...props.virtualScrollerOptions,
    ...{
      style: { ...props.virtualScrollerOptions.style, ...{ height: `${calc}px` } },
      autoSize: true,
      items: items,
      itemTemplate: (item, options) => item && createItem(item, options.index, options),
      contentTemplate: (options) => {
      const listProps = mergeProps(
        {
          id: props.listId,
          ref: options.contentRef,
          style: options.style,
          className: cx('list', { virtualScrollerProps, options }),
          role: 'listbox'
        },
        _ptm('list')
      );
      
      return <ul {...listProps}>{options.children}</ul>;
    }
  }
};
return <VirtualScroller ref={virtualScrollerRef} {...virtualScrollerProps} pt={_ptm('virtualScroller')} __parentMetadata={{ parent: props.metaData }} />;
}
```

VirtualScroller 컴포넌트는 
Size of the viewport is configured usingscrollWidth, scrollHeight properties directly or with CSS width and height styles.
으로 configure하고 있으며 아이템 수의 상관 없이 고정된 사이즈를 유지합니다.

이슈에서 의도하고자 하는 동작을 하기 위해선 VirtualScroller 컴포넌트의 수정이 필요해 보이는데, VirtualScroller를 사용하는 많은 컴포넌트들을 생각하면 과연 맞는지 모르겠습니다.

아니라면 추가적으로 doc을 업데이트하여 height에 대한 resize가 안된다는 것을 알리는 방법도 있다고 생각합니다.

**Comment by blan19** [comment]
AutoComplete component's Virtual Scroll internally uses VirtualScroller component.

```typescript
if (props.virtualScrollerOptions) {
  // ... rest
  return <VirtualScroller ref={virtualScrollerRef} {...virtualScrollerProps} pt={_ptm('virtualScroller')} __parentMetadata={{ parent: props.metaData }} />;
}
```

`Size of the viewport is configured usingscrollWidth, scrollHeight properties directly or with CSS width and height styles.`

VirtualScroller component appears to maintain a fixed size regardless of the number of items. 

To achieve the behavior intended in issue, it seems that modifications to VirtualScroller component might be necessary. 

However, considering the many components that use VirtualScroller, I'm not certain if this is the right approach.

**Comment by blan19** [comment]
AutoComplete component's Virtual Scroll internally uses VirtualScroller component.

```typescript
if (props.virtualScrollerOptions) {
  // ... rest
  return <VirtualScroller ref={virtualScrollerRef} {...virtualScrollerProps} pt={_ptm('virtualScroller')} __parentMetadata={{ parent: props.metaData }} />;
}
```

`Size of the viewport is configured usingscrollWidth, scrollHeight properties directly or with CSS width and height styles.`

VirtualScroller component appears to maintain a fixed size regardless of the number of items

therefore, I am not certain whether this is a bug or not..

@melloware 

to achieve the behavior intended in issue, it seems that modifications to VirtualScroller component might be necessary

however, considering the many components that use VirtualScroller, I'm not certain if this is the right approach..

If not, another option could be to update the documentation to inform users that resizing is not supported

thank you xD

**Comment by melloware** [comment]
Agreed the VirtualScroller is used throughout the library and three are various bugs reported against it but its a PrimeTek core component they use in many libraries so I am hestitant to touch it. I usually leave VirtualScroller issues to PrimeTek.

**Comment by inad9300** [comment]
Still a bug.
